src/tags/object/Paragraphs/Paragraphs.js

```diff
@@ -286,14 +302,56 @@ class HtxParagraphsView extends Component {
     }
   }
 
+  /**
+   * Generates a textual representation of the current selection range.
+   * Excludes any invalid nodes, and their children. This is necessary to avoid
+   * capturing the text of elements such as the author name.
+   *
+   * @param {string} textClass 
+   * @param {Text[]} phrases
+   * @param {number} start
+   * @param {number} end
+   * @param {number} startOffset
+   * @param {number} endOffset
+   * @returns {string}
+   */
+  _getResultText(textClass, phrases, range, start, startOffset, endOffset) {
+    const nodesInRange = getTextNodesInRange(range);
+    const endNodeIndex = nodesInRange.length - 1;
+
+    const resultText = nodesInRange
+      .reduce((acc, _n, i) => {
+        const str = phrases[+start + i]
```

Comment:
I'm not sure here... This implies, that these returned text nodes will be perfectly aligned with phrases, but that doesn't feel true, because if we are filtering them out there are more text nodes than phrases in the range.

Reply:
We are clearly looking here for text content of the whole phrase elements, so we can just calculate indices of start and end phrase, collect their texts and cut off by start/endOffsets. With no NodesInRange at all, because we actually don't need them in such easy case as Paragraphs.

Reply:
Am I missing something?

Reply:
Nope you are correct, this is now updated to using the same phraseElements, and simplified to just reuse the indices we already have.